bndtools.core/test/org/bndtools/core/editors/BuildpathQuickFixProcessorTest.java

```diff
@@ -0,0 +1,12 @@
+package org.bndtools.core.editors;
+
+import org.junit.Test;
+
+public class BuildpathQuickFixProcessorTest {
+
+	@Test
+	public void testBasic() {
+		// not sure how to test this
+		// the majority of the code is tested in bnd
+	}
```

Comment:
I don't think it's accurate to say that "the majority of the code is tested in bnd". I'm sure that all of the *searching* code is tested in Bnd - but searching is only half of the quick fix processor's responsibility. The other thing that the quick fix processor has to do (equally important) is understand Eclipse's error message formats so that it knows what to search for and can translate that into a form that Bnd understands, and then conversely take Bnd's response and translate that into a form that Eclipse can use. The first part in particular is a non-trivial operation and accounted for a significant chunk of the original quick-fix code.

The original `ImportPackageQuickFixProcessorTest` was close to a full end-to-end test - it used actual source code with errors which could be quick-fixed, then it used Eclipse's compiler to compile these into an actual AST, and finally it created actual `IProblemLocation` objects (which were built based on observed values determined by experimentation on a live Eclipse instance). It passed these in to `getCompletions()` to ensure that the correct set of `ICompletionProposal`s was created.

In order to ensure that we haven't had any regressions, we should do something similar. The old test should form the basis, as I know it had pretty good coverage of lots of weird corner cases. Your code is admittedly a lot simpler, but I'm not sure if this simplification has come at the expense of us not handling some of these weird corner cases.

The more I think about this as I type, the more I am of the opinion that the best way forward for me to review your changes will be to port `ImportPackageQuickFixProcessor` to `BuildpathQuickFixProcessorTest`. For starters, I can't remember all of the weird corner cases, and even if I could testing them all manually would be onerous. If I instead go to the effort of porting the test, then we have the added benefit of a solid regression test moving forward so that others can make changes without fear of breaking existing functionality (that is, after all, why I went to the effort of writing the original regression test in the first place!)

Reply:
That was what I was secretly hoping for :-) 

It is always very nice if you have a different person doing the test.

Reply:
If you just could have a standalone test where I can give a Java source with a problem and get the quick fixes, that would be great. We could use this in other cases as well since this stuff is highly undertested.

Reply:
I'll have a look when I get a chance.